pynicotine/plugins/core_commands/__init__.py

```diff
@@ -26,39 +26,637 @@ def __init__(self, *args, **kwargs):
+                "group": _("Now Playing"),
+            },
+            "add": {
+                "aliases": ["buddy"],
+                "callback": self.add_buddy_command,
+                "description": _("Add user to buddy list"),
+                "group": _("Users"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "rem": {
+                "aliases": ["unbuddy"],
+                "callback": self.remove_buddy_command,
+                "description": _("Remove user from buddy list"),
+                "group": _("Users"),
+                "usage": ["<buddy>"],
+                "usage_private_chat": ["[buddy]"]
+            },
+            "ban": {
+                "callback": self.ban_user_command,
+                "description": _("Stop connections from user"),
+                "group": _("Users"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "unban": {
+                "callback": self.unban_user_command,
+                "description": _("Remove user from ban list"),
+                "group": _("Users"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "block": {  # new
+                "callback": self.block_user_ip_command,
+                "description": _("Stop all connections from same IP as user"),
+                "group": _("Network Filters"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "unblock": {  # new
+                "callback": self.unblock_user_ip_command,
+                "description": _("Remove user's IP address from block list"),
+                "group": _("Network Filters"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "ignore": {
+                "callback": self.ignore_user_command,
+                "description": _("Silence chat messages from user"),
+                "disable": ["cli"],
+                "group": _("Users"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "unignore": {
+                "callback": self.unignore_user_command,
+                "description": _("Remove user from chat ignore list"),
+                "disable": ["cli"],
+                "group": _("Users"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "ignoreip": {
+                "callback": self.ignore_user_ip_command,
+                "description": _("Silence chat messages from IP address of user"),
+                "disable": ["cli"],
+                "group": _("Network Filters"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "unignoreip": {  # new
```

Comment:
unignore-ip

Reply:
The name of this new experimental command is based on the same naming convention as historically used.

It will probably be removed for now because implementing it requires significant effort that is beyond the scope of this PR.

Reply:
Let's hope our commands remain short enough then.

Reply:
Agreed, it's not reasonable to expect the user to deal with a whole range of commands each with subtle differences.

It is going to be necessary to allow for more flexible arguments, even if this making some assumptions, such as:

- If we specify an `ip_address`, then the intention is to ignore/unignore all usernames associated with that address.
- If we specify a `username`, then the intention may be inferred also to ignore/unignore all users at that same address.

The `<user>` argument now validates on either a "username" or an "ip_address" type of input for both the `/ignore` and `/unignore` commands, but a core function is missing from networkfilter.py to do unignore by ip address.

Reply:
Resolved, both username-based and IP-based ignore rules now get removed with a single `/unignore` command.